### ISSUE TITLE
[Bug]: \Pimcore\Cache\Core\WriteLock generates a lot of cache misses, skewing cache hit rate metrics

### ISSUE THREAD

**Issue by dkarlovi** [pimcore/pimcore#13102]
### Expected behavior

During regular operations, the write lock would cause no misses.

### Actual behavior

During regular operations, the write lock causes a lot of misses. For example, just refreshing the admin login screen (locally, no concurrent requests) it causes 8 misses. This means Pimcore has a baseline cache miss rate, making it more difficult to use standard metrics tools which use hit/miss rates.

### Steps to reproduce

1. switch to Redis cache
2. enable [Redis keyspace notifications](https://redis.io/docs/manual/keyspace-notifications/), like so (needs Redis 6 for `MISS` event)
```
redis-cli config set notify-keyspace-events Km
```
3. monitor key events:
```
redis-cli --csv psubscribe '__key*__:*'
```

Example output:
```
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:\x00tags\x00system_cache_write_lock","keymiss"
"pmessage","__key*__:*","__keyspace@0__:+3G7jg6MJe:system_cache_write_lock","keymiss"
```

**Comment by dvesh3** [comment]
@dkarlovi could you please check https://github.com/pimcore/pimcore/pull/13196? thanks!

**Comment by dkarlovi** [comment]
@dvesh3 I'll have time to test it Wednesday, is that OK?

**Comment by dvesh3** [comment]
@dkarlovi no rush. the release is scheduled for next week 😉

**Comment by dkarlovi** [comment]
@dvesh3 I can confirm this indeed fixes the issue. \o/ I can't say if the fix introduces any unwanted side-effects, but the issue described is indeed fixed.

**Comment by dvesh3** [comment]
@dkarlovi thanks for confirming 👍  we'll test it further to check side effects.

**Comment by brusch** [comment]
Fixed by #13196
